salt/cloud/clouds/vmware.py

```diff
@@ -471,20 +473,29 @@ def _add_new_scsi_controller_helper(scsi_controller_label, properties, bus_numbe
     return scsi_spec
 
 
-def _add_new_ide_controller_helper(ide_controller_label, properties, bus_number):
+def _add_new_ide_controller_helper(ide_controller_label, controller_key, bus_number):
     '''
     Helper function for adding new IDE controllers
 
     .. versionadded:: 2016.3.0
+
```

Comment:
Ok, I feel that this is not the best approach for 2 reasons:

First: most of the helpers surrounding the creation of device specs generate their own controller keys; it seems to be only in this specific case - so far - that a key should not be generated. So I chose to document this particular function to make the difference even more clear.

Second: I'd argue that choosing to not document internal functions because no other function has been is a little contradictory and simply encourages the situation to persist. 

However, say the word (again) and I shall remove it.